Feature/nav panel

harshita-jaiswal opened the pull request:
Changes Made

- Modified the Layout 
- Added Routing
- Added Navigation Panel and its nav items
- Change project structure

Reasons

Changes are made to add a navigation panel and structure the project.

Feedback

Does the UI look fine? Is there anything I am missing in the navigation panel? Does the project structure make sense?

JoshuaWachana commented on NBA-Stats-Dashboard/src/index.css:
```diff
@@ -41,6 +43,30 @@
+  color: var(--white);
+  opacity: 0.9;
+}
+.navItem:hover {
+  opacity: 1;
+}
```
This hover effect is quite subtle and easy to miss to go from opacity 0.9 to 1. We should consider things like changing the background color and text color of the navItem and such, just so that it is super obvious when each item is hovered over.

harshita-jaiswal replied:
Good idea! I've made changes to the CSS, but I think we can make the interface even better. Want to talk about the design?

JoshuaWachana replied:
Yes, we will talk more about design (styling) later. For the next few PRs the functionality will be more important.

JoshuaWachana commented on NBA-Stats-Dashboard/src/main.jsx:
```diff
@@ -2,9 +2,12 @@ import React from 'react';
 import ReactDOM from 'react-dom/client';
 import App from './App.jsx';
 import './index.css';
+import { BrowserRouter as Router } from "react-router-dom";
```
This line gave me an issue on my end even after installing all the dependencies using "npm i". I don't think this dependency was added to the package.json

harshita-jaiswal replied:
The problem's sorted. I mistakenly installed the package in the main directory, creating extra node_modules and package.json files there. I overlooked adding a package to the main package.json while cleaning up.

JoshuaWachana reviewed (commented):
Hello Harshita, the UI looks fine, and so does the code for the navigational menu. Thanks for setting up the routes and editing the project structure.

However, there is just one small problem. Originally I had a problem getting the app to start because it seems like "react-router-dom" was not added to the dev dependencies in the package.json.

JoshuaWachana commented:
SInce it looks like you added a dependency "react-router-dom" in this PR, but there are no new changes to the package.json file it seems to me that the issue was that you forgot to push it or that you did not install it with the "--save".

Below are the issues I got when I originally tried to run the app.

![err1](https://github.com/JoshuaWachana/NBA-Stats-Dashboard/assets/38127011/81462dd5-f2a7-4257-9a19-a04ada0bd725)
![err1b](https://github.com/JoshuaWachana/NBA-Stats-Dashboard/assets/38127011/52385ebc-4904-4e45-9306-3235c0ead0d1)
![err2](https://github.com/JoshuaWachana/NBA-Stats-Dashboard/assets/38127011/866a9943-e770-4dfc-8c52-632b27b079db)

Below is what the package.json looks like, as you can see "react-router-dom" is not listed as a dependency

![pj1](https://github.com/JoshuaWachana/NBA-Stats-Dashboard/assets/38127011/5c4a10e5-1d1d-4100-a7af-969d5ce0a2aa)

So I opened up another terminal and I fixed that with "npm install --save react-router-dom" 

![sol1](https://github.com/JoshuaWachana/NBA-Stats-Dashboard/assets/38127011/a1d68f0f-8842-45c4-847b-0e776917ac12)

Here is how the package json looks after, "react-router-dom" is now listed as a dependency.

![pj2](https://github.com/JoshuaWachana/NBA-Stats-Dashboard/assets/38127011/634c18c5-57dd-4ac3-bb8a-79aa195c0cc0)

Then it ran smoothly after a quick refresh.

JoshuaWachana commented:
Also, please change the file name from "REAME.md" to "README.md"

The "D" was missing, and this is why a GitHub user can not automatically view it.

harshita-jaiswal commented:
> Also, please change the file name from "REAME.md" to "README.md"
> 
> The "D" was missing, and this is why a GitHub user can not automatically view it.

Done

harshita-jaiswal commented:
> SInce it looks like you added a dependency "react-router-dom" in this PR, but there are no new changes to the package.json file it seems to me that the issue was that you forgot to push it or that you did not install it with the "--save".
> 
> Below are the issues I got when I originally tried to run the app.
> 
> ![err1](https://user-images.githubusercontent.com/38127011/284720167-81462dd5-f2a7-4257-9a19-a04ada0bd725.png) ![err1b](https://user-images.githubusercontent.com/38127011/284720169-52385ebc-4904-4e45-9306-3235c0ead0d1.png) ![err2](https://user-images.githubusercontent.com/38127011/284720171-866a9943-e770-4dfc-8c52-632b27b079db.png)
> 
> Below is what the package.json looks like, as you can see "react-router-dom" is not listed as a dependency
> 
> ![pj1](https://user-images.githubusercontent.com/38127011/284720510-5c4a10e5-1d1d-4100-a7af-969d5ce0a2aa.png)
> 
> So I opened up another terminal and I fixed that with "npm install --save react-router-dom"
> 
> ![sol1](https://user-images.githubusercontent.com/38127011/284720959-a1d68f0f-8842-45c4-847b-0e776917ac12.png)
> 
> Here is how the package json looks after, "react-router-dom" is now listed as a dependency.
> 
> ![pj2](https://user-images.githubusercontent.com/38127011/284721167-634c18c5-57dd-4ac3-bb8a-79aa195c0cc0.png)
> 
> Then it ran smoothly after a quick refresh.

Fixed

harshita-jaiswal commented:
> Hello Harshita, the UI looks fine, and so does the code for the navigational menu. Thanks for setting up the routes and editing the project structure.
> 
> However, there is just one small problem. Originally I had a problem getting the app to start because it seems like "react-router-dom" was not added to the dev dependencies in the package.json.

Done

JoshuaWachana reviewed (approved):
It works well now. Thanks for addressing the feedback.
